src/qat/model/hardware_model.py

```diff
@@ -58,30 +62,66 @@ class PhysicalHardwareModel(LogicalHardwareModel):
 
     :param qubits: The superconducting qubits on the chip.
     :param physical_connectivity: The connectivities of the physical qubits on the QPU.
+    :param physical_connectivity_quality: Quality of the connections between the qubits.
     :param logical_connectivity: The connectivities of the qubits used for compilation,
                     which is equal to `physical_connectivity` or a subset thereof.
     """
 
     qubits: dict[QubitId, Qubit]
     physical_connectivity: dict[QubitId, set[QubitId]] = Field(frozen=True)
     logical_connectivity: Optional[dict[QubitId, set[QubitId]]] = Field(default=None)
+    physical_connectivity_quality: dict[tuple[QubitId, QubitId], CalibratableUnitInterval]
 
     @model_validator(mode="before")
-    def validate_connectivity(cls, data):
-        physical_connectivity = data["physical_connectivity"]
-        logical_connectivity = data.get("logical_connectivity", None)
-
-        if not logical_connectivity:
+    def default_logical_connectivity(cls, data):
+        if not data.get("logical_connectivity", None):
             logical_connectivity = deepcopy(data["physical_connectivity"])
             data["logical_connectivity"] = logical_connectivity
 
-        for qubit_index in logical_connectivity:
-            if not logical_connectivity[qubit_index] <= physical_connectivity[qubit_index]:
+        return data
+
+    @model_validator(mode="before")
+    def default_physical_connectivity_quality(cls, data):
+        if not data.get("physical_connectivity_quality", None):
+            physical_connectivity_quality = {}
+
+            for q1_index, connected_qubits in data["physical_connectivity"].items():
+                for q2_index in connected_qubits:
+                    physical_connectivity_quality[(q1_index, q2_index)] = 1.0
+
+            data["physical_connectivity_quality"] = physical_connectivity_quality
+
+        return data
+
+    @model_validator(mode="after")
+    def validate_connectivity(self):
+        for qubit_index in self.logical_connectivity:
+            if (
+                not self.logical_connectivity[qubit_index]
+                <= self.physical_connectivity[qubit_index]
+            ):
                 raise ValueError(
                     "Logical connectivity must be a subgraph of the physical connectivity."
                 )
 
-        return data
+        return self
+
+    @model_validator(mode="after")
+    def validate_connectivity_quality(self):
+        for q1_index, q2_index in self.physical_connectivity_quality:
+            if not self.physical_connectivity.get(q1_index, None):
```

Comment:
I assume that the quality of the physical connectivity/coupling between the qubits is directional as well? I.e., `quality(qubit1, qubit2) != quality(qubit2, qubit1)`?

Can someone confirm this?

Reply:
I agree with this! 

Out of curiosity, what decides the value of the quality? Is it just the measured fidelity of the 2q gate?

Reply:
cc/ @bryn-bell @elskysr